L1Trigger/TrackFindingTracklet/src/HybridFit.cc

```diff
@@ -195,55 +195,108 @@ void HybridFit::Fit(Tracklet* tracklet, std::vector<const Stub*>& trackstublist)
                                   << ", phi0 = " << trk.phi0() << ", eta = " << trk.eta() << ", z0 = " << trk.z0()
                                   << ", chi2 = " << trk.chi2() << ", accepted = " << trk.accepted();
 
-    // Tracklet wants phi0 with respect to lower edge of sector, not global phi0.
-    double phi0fit = reco::reduceRange(trk.phi0() - iSector_ * 2 * M_PI / N_SECTOR + 0.5 * settings_.dphisectorHG());
-
-    double rinvfit = 0.01 * settings_.c() * settings_.bfield() * trk.qOverPt();
-
-    int irinvfit = rinvfit / settings_.krinvpars();
-    int iphi0fit = phi0fit / settings_.kphi0pars();
-    int itanlfit = trk.tanLambda() / settings_.ktpars();
-    int iz0fit = trk.z0() / settings_.kz0pars();
-    int id0fit = trk.d0() / settings_.kd0pars();
-    int ichi2rphifit = trk.chi2rphi() / 16;
-    int ichi2rzfit = trk.chi2rz() / 16;
-
-    const vector<const tmtt::Stub*>& stubsFromFit = trk.stubs();
-    vector<const L1TStub*> l1stubsFromFit;
-    for (const tmtt::Stub* s : stubsFromFit) {
-      unsigned int IDf = s->index();
-      const L1TStub* l1s = L1StubIndices.at(IDf);
-      l1stubsFromFit.push_back(l1s);
-    }
+    if (trk.done_bcon()) {
```

Comment:
Can you eliminate all the duplicate code between the done_bcon and not done_bcon options? e.g. How about defining float phi0fit, rinvfit, d0fit & chi2rphi, and setting them to bcon or non-bcon values, according to the value of done_bcon. Then these variables can be passed to setFitPars().

Reply:
I have made changes to reflect this, let me know if this is what you were invisioning.

Reply:
Yes, almost. Though you should be able to eliminate also duplicate code at https://github.com/cgsavard/cmssw/blob/add_beam_constr/L1Trigger/TrackFindingTracklet/src/HybridFit.cc#L206 and https://github.com/cgsavard/cmssw/blob/add_beam_constr/L1Trigger/TrackFindingTracklet/src/HybridFit.cc#L215 .

Reply:
Both of these lines rely on the trk phi0 which has a bcon version so this will be different with the beam spot constraint turned on

Reply:
Can't you do:

if (trk.done_bcon()) {
   phi0fit = trk.phi0_bcon();
   ...
} else {
  phi0fit = trk.phi0();
  ...
}
phi0fit = reco::reduceRange(phi0fit - iSector_ * 2 * M_PI / N_SECTOR + 0.5 * settings_.dphisectorHG());
...

Reply:
I see. In this care then I should also do this to rinvfit I would presume? It would be the same thing with 

if (trk.done_bcon()) {
qoverpt = trk.qOverPt_bcon();
...
} else {
qoverpt = trk.qOverPt();
...
}
rinvfit = 0.01 * settings_.c() * settings_.bfield() * qoverpt;
...

Reply:
I made these changes